### PR TITLE
Balder/pack wrappers tighter

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

File: storage/src/vespa/storage/bucketdb/abstract_bucket_map.h [hunk 1]
```diff
@@ -28,47 +29,69 @@ class AbstractBucketMap {
...
+            : _lockKeeper(map, key),
               _value(),
-              _clientId(clientId) {}
+              _clientId(clientId),
+              _exists(false),
+              _preExisted(false) {}
         // TODO noexcept on these:
-        WrappedEntry(WrappedEntry&&) = default;
-        WrappedEntry& operator=(WrappedEntry&&) = default;
+        WrappedEntry(WrappedEntry&&) noexcept = default;
```
Comment (vekterli) [inline storage/src/vespa/storage/bucketdb/abstract_bucket_map.h:93]: Consider removing the now done TODO comment

File: storage/src/vespa/storage/bucketdb/abstract_bucket_map.h [hunk 2]
```diff
@@ -28,47 +29,69 @@ class AbstractBucketMap {
...
+            rhs._map = nullptr;
+            return *this;
+        }
+        ~LockKeeper() { cleanup(); }
+        AbstractBucketMap & map() { return *_map; }
+        void unlock() {
+            _map->unlock(_key);
+            _map = nullptr;
+        }
+        bool locked() const noexcept { return _map != nullptr; }
```
Comment (vekterli) [inline storage/src/vespa/storage/bucketdb/abstract_bucket_map.h:60]: Consider adding a `[[nodiscard]]` attribute here

File: storage/src/vespa/storage/bucketdb/abstract_bucket_map.h [hunk 3]
```diff
@@ -11,6 +11,7 @@
 #include <functional>
 #include <iosfwd>
 #include <map>
+#include <optional>
```
Comment (vekterli) [inline storage/src/vespa/storage/bucketdb/abstract_bucket_map.h:14]: Is this needed or a leftover?

**Review by vekterli — DISMISSED** [review]
:+1:

File: storage/src/vespa/storage/bucketdb/abstract_bucket_map.h (hunk 3 above)
Comment (baldersheim) [reply storage/src/vespa/storage/bucketdb/abstract_bucket_map.h]: leftover, thanks.

**Pull request merged**